pkg/csi/service/vanilla/controller.go

```diff
@@ -446,13 +501,50 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 	if err != nil {
 		return nil, err
 	}
+	var volumePath string
+
+	// in-tree volume support
+	var isVolumePath bool
+	var volumeID string
+	if strings.Contains(req.VolumeId, ".vmdk") {
+		volumePath = req.VolumeId
+		isVolumePath = true
```

Comment:
Can we avoid isVolumePath? If volumePath is set, can we not assume that we are dealing with a in tree PV?

Reply:
yes we can rely on `volumePath`. Removed usage of `isVolumePath`.